File: backend/API/Controllers/BaseApiController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class BaseApiController : ControllerBase
    {
        [HttpGet("not-found")]
        public ActionResult<string> GetNotFound()
        {
            return NotFound();
        }

        [HttpGet("bad-request")]
        public ActionResult<string> GetBadRequest()
        {
            return BadRequest(new ProblemDetails{Title = "This was not a good request"});
        }

        [HttpGet("server-error")]
        public ActionResult<string> GetServerError()
        {
            return StatusCode(500, "This is a server error");
        }

        [HttpGet("unauthorized")]
        public ActionResult<string> GetUnauthorized()
        {
            return Unauthorized();
        }

        [HttpGet("validation-error")]
        public ActionResult<string> GetValidationError()
        {
            ModelState.AddModelError("Problem1", "This is the first error");
            ModelState.AddModelError("Problem2", "This is the second error");
            return ValidationProblem();
        }
    }
}

File: backend/MAIN_API/src/base-api/base-api.controller.ts
import { BadRequestException, Controller, Get, HttpCode, HttpStatus, NotFoundException, UnauthorizedException } from '@nestjs/common';
import { problem, validationProblem } from '../common/problem-details';

@Controller('baseApi')
export class BaseApiController {
  @Get('not-found') notFound() { throw new NotFoundException(); }
  @Get('bad-request') badRequest() { throw problem('This was not a good request'); }
  @Get('server-error') @HttpCode(HttpStatus.INTERNAL_SERVER_ERROR) serverError() { return 'This is a server error'; }
  @Get('unauthorized') unauthorized() { throw new UnauthorizedException(); }
  @Get('validation-error') validationError() {
    throw validationProblem({ Problem1: ['This is the first error'], Problem2: ['This is the second error'] });
  }
}
